config/routes.rb

```diff
@@ -1,9 +1,12 @@
 Rails.application.routes.draw do
   root 'bars#index'
   devise_for :users
+  get 'users/:id' => 'users#destroy', :via => :delete,
+                                      :as => :admin_destroy_user
+  get 'users/:id' => 'users#show', as: :user
```

Comment:
I'm not sure all this is necessary. Take a look at [Game Collector](https://horizon.launchacademy.com/lessons/namespacing-with-tdd) which covers admin namespacing in some more detail.